src/game/client/draw/w3dprojectilestreamdraw.cpp

```diff
@@ -0,0 +1,209 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief W3D Projectile Stream Draw Module
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "w3dprojectilestreamdraw.h"
+#include "assetmgr.h"
+#include "drawable.h"
+#include "object.h"
+#include "projectilestreamupdate.h"
+#include "segline.h"
+#include "texture.h"
+#include "w3ddisplay.h"
+#include "w3dscene.h"
+
+W3DProjectileStreamDrawModuleData::W3DProjectileStreamDrawModuleData() :
+    m_width(0.0f), m_tileFactor(0.0f), m_scrollRate(0.0f), m_maxSegments(0)
+{
+}
+
+void W3DProjectileStreamDrawModuleData::Build_Field_Parse(MultiIniFieldParse &p)
+{
+    // clang-format off
+    static FieldParse dataFieldParse[] = {
+        { "Texture", &INI::Parse_AsciiString, nullptr, offsetof(W3DProjectileStreamDrawModuleData, m_texture) },
+        { "Width", &INI::Parse_Real, nullptr, offsetof(W3DProjectileStreamDrawModuleData, m_width) },
+        { "TileFactor", &INI::Parse_Real, nullptr, offsetof(W3DProjectileStreamDrawModuleData, m_tileFactor) },
+        { "ScrollRate", &INI::Parse_Real, nullptr, offsetof(W3DProjectileStreamDrawModuleData, m_scrollRate) },
+        { "MaxSegments", &INI::Parse_Int, nullptr, offsetof(W3DProjectileStreamDrawModuleData, m_maxSegments) },
+        { nullptr, nullptr, nullptr, 0 },
+    };
+    // clang-format on
+
+    ModuleData::Build_Field_Parse(p);
+    p.Add(dataFieldParse, 0);
+}
+
+W3DProjectileStreamDraw::W3DProjectileStreamDraw(Thing *thing, ModuleData const *module_data) :
+    DrawModule(thing, module_data)
+{
+    m_texture = W3DAssetManager::Get_Instance()->Get_Texture(Get_W3D_Projectile_Stream_Draw_Module_Data()->m_texture);
+
+    for (int i = 0; i < 20; i++) {

```

Comment:
Either use a constant or `ARRAY_SIZE(m_allLines)` for all the 20's in this class.

Reply:
Used ARRAY_SIZE for this.